torch/lib/THS/generic/THSTensor.c

```diff
@@ -329,20 +481,26 @@ void THSTensor_(reorder)(THSTensor *self) {
   for (j = 1; j < self->nnz; j++) {
     cmp = 1;
     // TODO: pass eps in as a parameter
-    if (values[j] == 0) continue;
-    for (d = 0; d < ndim; d++)
+    // if (values[j] == 0) continue;
+    for (d = 0; d < nDimI; d++)
       if (IND(i, d) != IND(j, d)) {
         cmp = 0;
         break;
       }
-    if (cmp) values[i] += values[j];
+    if (cmp) {
+      THSTensor_(addSlice)(dstBuffer, dstBuffer, srcBuffer, values_, values_, 1, values_, 0, i, i, j);
```

Comment:
nit: It might be more readable to allow passing in `NULL`s as one of the source buffers.